pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap.go

```diff
@@ -111,84 +104,98 @@ func DoBootstrap(
 	if configSpec != nil {
 		err := doReconfigure(vmCtx, vcVM, configSpec)
 		if err != nil {
-			return fmt.Errorf("boostrap reconfigure failed: %w", err)
+			return fmt.Errorf("bootstrap reconfigure failed: %w", err)
 		}
 	}
 
 	if customSpec != nil {
 		err := doCustomize(vmCtx, vcVM, config, customSpec)
 		if err != nil {
-			return fmt.Errorf("boostrap customize failed: %w", err)
+			return fmt.Errorf("bootstrap customize failed: %w", err)
 		}
 	}
 
 	return nil
 }
 
-func getBootstrapArgs(
-	vmCtx context.VirtualMachineContextA2,
-	k8sClient ctrl.Client,
-	isCloudInit, isGOSC bool,
+// GetBootstrapArgs returns the information used to bootstrap the VM via
+// one of the many, possible bootstrap engines.
+func GetBootstrapArgs(
+	ctx context.VirtualMachineContextA2,
+	klient ctrl.Client,
```

Comment:
This name makes me think of `Kris Kross`. `k8sClient` isn't that more verbose :)